src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -0,0 +1,159 @@
+            Assert.Throws<ArgumentOutOfRangeException>("month", () => calendar.GetDaysInMonth(yearOffset + 1, calendar.MaxSupportedDateTime.Month + 1, 1));
+
+            // Era is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetDaysInMonth(yearOffset + 1, 10, -1));
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetDaysInMonth(yearOffset + 1, 10, MaxEra(calendar) + 1));
+        }
+
+        public static void GetMonthsInYear_Invalid(Calendar calendar)
+        {
+            int yearOffset = YearOffset(calendar);
+
+            // Year is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetMonthsInYear(-1, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetMonthsInYear(0, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetMonthsInYear(calendar.MinSupportedDateTime.Year + yearOffset - 1, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetMonthsInYear(calendar.MaxSupportedDateTime.Year + yearOffset + 1, 1));
+            
+            // Era is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetMonthsInYear(yearOffset + 1, -1));
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetMonthsInYear(yearOffset + 1, MaxEra(calendar) + 1));
+        }
+
+        public static void GetWeekOfYear_Invalid(Calendar calendar)
+        {
+            // Rule is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("rule", () => calendar.GetWeekOfYear(calendar.MaxSupportedDateTime, CalendarWeekRule.FirstDay - 1, DayOfWeek.Saturday));
+            Assert.Throws<ArgumentOutOfRangeException>("rule", () => calendar.GetWeekOfYear(calendar.MaxSupportedDateTime, CalendarWeekRule.FirstFourDayWeek + 1, DayOfWeek.Saturday));
+
+            // FirstDayOfWeek is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("firstDayOfWeek", () => calendar.GetWeekOfYear(calendar.MaxSupportedDateTime, CalendarWeekRule.FirstDay, DayOfWeek.Sunday - 1));
+            Assert.Throws<ArgumentOutOfRangeException>("firstDayOfWeek", () => calendar.GetWeekOfYear(calendar.MaxSupportedDateTime, CalendarWeekRule.FirstDay, DayOfWeek.Saturday + 1));
+        }
+
+        public static void ToDateTime_Invalid(Calendar calendar)
+        {
+            int yearOffset = YearOffset(calendar);
+
+            DateTime maxDate = new ThaiBuddhistCalendar().MaxSupportedDateTime;
+            DateTime minDate = new ThaiBuddhistCalendar().MinSupportedDateTime;
+
+            // Year is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => new ThaiBuddhistCalendar().ToDateTime(-1, minDate.Month, minDate.Day, minDate.Hour, minDate.Minute, minDate.Second, minDate.Millisecond, 1));
```

Comment:
Why is ```1``` for the era hardcoded in all of these?  Shouldn't that instead be pulled from the calendar?